### ISSUE TITLE
Integrate Microsoft login from Firebase 

### ISSUE THREAD

**Issue by adarshanand67** [adarshanand67/ShareMe#17]
Github and google already done

**Comment by adarshanand67** [comment]
Microsoft login from Firebase - https://firebase.google.com/docs/auth/web/microsoft-oauth

**Comment by shivanshi-s** [comment]
hey, im a part of jwoc and i would love to take up this issue and give it a try!

**Comment by adarshanand67** [comment]
> hey, im a part of jwoc and i would love to take up this issue and give it a try!

Alright assigning the issue

**Comment by adarshanand67** [comment]
@shivanshi-s are you interested?

**Comment by shivanshi-s** [comment]
> @shivanshi-s are you interested?

yes i am! trying it out, give me one more day and ill let u know!

**Comment by shivam1317** [comment]
Can i work on this issue?

**Comment by adarshanand67** [comment]
> Can i work on this issue?
@sandeeprawat28 you up for this?

**Comment by adarshanand67** [comment]
By going to Firebase console you can add Microsoft option as well
![image](https://user-images.githubusercontent.com/73928744/219609605-90571616-b0f8-413b-a606-04bb1cf88366.png)
Create a new button just like Google,Github and have Microsoft auth @shivam1317

**Comment by adarshanand67** [comment]
Let me know if you need API Keys, callback URLs

**Comment by shivam1317** [comment]
> By going to Firebase console you can add Microsoft option as well
> ![image](https://user-images.githubusercontent.com/73928744/219609605-90571616-b0f8-413b-a606-04bb1cf88366.png)
> Create a new button just like Google,Github and have Microsoft auth @shivam1317 

You have to enable Microsoft login in your console because I don't have access of it

**Comment by shivam1317** [comment]
Hey @adarshanand67 Can you enable the microsoft login from your console.

**Comment by shivam1317** [comment]
hey @adarshanand67, have you enabled microsoft login?

**Comment by adarshanand67** [comment]
Not yet, if possible can you do it from your side?

**Comment by adarshanand67** [comment]
> > By going to Firebase console you can add Microsoft option as well
> > ![image](https://user-images.githubusercontent.com/73928744/219609605-90571616-b0f8-413b-a606-04bb1cf88366.png)
> > Create a new button just like Google,Github and have Microsoft auth @shivam1317
> 
> You have to enable Microsoft login in your console because I don't have access of it

How can I register an app?

**Comment by shivam1317** [comment]
> Not yet, if possible can you do it from your side?

i can't do it as i don't have access to your firebase account.

To enable microsoft login in your firebase console follow this:
Goto firebase console -> authentication -> sign-in method -> click on add new provider

![image](https://user-images.githubusercontent.com/70281451/219855587-34a86e99-afec-49b3-b5ae-aa158594fff1.png)

Add `application id` and `application secret` and click `save`

![image](https://user-images.githubusercontent.com/70281451/219855565-67144465-b1a9-4787-b45f-5c74f148393f.png)

**Comment by adarshanand67** [comment]
And where to get application id and application secret?

**Comment by adarshanand67** [comment]
Doing for Github was easy need to check for Microsoft

**Comment by shivam1317** [comment]
> And where to get application id and application secret?

You can follow this: https://medium.com/nerd-for-tech/adding-microsoft-authentication-using-firebase-485f0bd1e3eb
